bundles/org.openhab.binding.bluetooth.listener/src/main/java/org/openhab/binding/bluetooth/listener/internal/ListenerHandler.java

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bluetooth.listener.internal;
+
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.Arrays;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import javax.measure.MetricPrefix;
+import javax.measure.quantity.Time;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.bluetooth.BeaconBluetoothHandler;
+import org.openhab.binding.bluetooth.BluetoothBindingConstants;
+import org.openhab.binding.bluetooth.BluetoothCharacteristic;
+import org.openhab.binding.bluetooth.notification.BluetoothScanNotification;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.util.HexUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ListenerHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author VitaTucek - Initial contribution
+ */
+@NonNullByDefault
+public class ListenerHandler extends BeaconBluetoothHandler {
+    private final Logger logger = LoggerFactory.getLogger(ListenerHandler.class);
+    private @Nullable ListenerConfiguration config;
+    private final AtomicBoolean receivedStatus = new AtomicBoolean();
+    private @NonNullByDefault({}) ScheduledFuture<?> heartbeatFuture;
```

Comment:
Don't hide null checks
```suggestion
    private @Nullable ScheduledFuture<?> heartbeatFuture;
```